### PR TITLE
Update PowerShell to build with .NET Core 2.0.5

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Use new .Net Core servicing build 2.0.5.
https://github.com/dotnet/cli/releases
https://blogs.msdn.microsoft.com/dotnet/2018/01/09/net-core-january-2018-update/

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [NA] Issue filed - Issue link:
- [NA] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Comment by markekraus** [comment]
Pleas push a commit with `[Feature]` in the commit message.

**Comment by bergmeister** [comment]
Do we need to mention the 2 important security fixes that come with 2.0.5 in the documents?
- https://github.com/dotnet/announcements/issues/52
- https://github.com/dotnet/announcements/issues/51
Does someone need to review their severity/impact whether this also means that 6.0 needs to have a patch with this update? @LeeHolmes

File: appveyor.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 image: Visual Studio 2017
 
-# cache version - netcoreapp.2.0.4
+# cache version - netcoreapp.2.0.5
```
Comment (bergmeister) [inline appveyor.yml]: Do we need to drop one of the caches (maybe in a separate PR)? I have seen recently that we always hit the 1GB of AppVeyor where it tries to upload it but then fails.

**Comment by bergmeister** [comment]
Should we also update the SDK in the [global.json](https://github.com/PowerShell/PowerShell/blob/master/global.json) from `2.0.2` to `2.0.3`, which is alrady in the used AppVeyor image [VisualStudio2017](https://www.appveyor.com/docs/build-environment/)?

File: appveyor.yml (hunk 1 above)
Comment (markekraus) [reply appveyor.yml]: touching the appveory.yml should cause the cache to rebuild.

**Comment by iSazonov** [comment]
I think could we move to latest SDK (2.1.4) after GA 6.0.1 to get up to date the 6.0.0 servicing branch.

Interesting, will we be able to move to CoreFX 2.1.0 preview version in our development branch - this version already contains many performance improvements and we could use new C# 7.2.

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@LeeHolmes @PaulHigin @SteveL-MSFT can you please take a look at the comment https://github.com/PowerShell/PowerShell/pull/5903#issuecomment-357796209 and see if there's any implication on PS Core 6.0.0?

**Review by TravisEz13 — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
We should take dotnetcore 2.0.5 as part of 6.0.1 release

**Comment by bergmeister** [comment]
@SteveL-MSFT OK. On a related note: The WiX toolset also had a security patch 2 weeks ago: https://www.firegiant.com/blog/2017/12/31/wix-v3.10.4-and-wix-v3.11.1-released/

**Comment by iSazonov** [comment]
It seems we use WiX from AppVeyor base image - so the issue is for AppVeyor.

**Comment by bergmeister** [comment]
Yes, but I guess that MSFT uses their own servers for building the GA release. This was a remark about the upcoming patch release.

**Comment by bergmeister** [comment]
@iSazonov @SteveL-MSFT I am confused, the [global.json](https://github.com/PowerShell/PowerShell/blob/master/global.json) still refers to `2.0.2` and the documentation on it [here](https://docs.microsoft.com/en-us/dotnet/core/tools/global-json) says that this is the SDK (not the runtime version). This SDK version contains the old 2.0.0. version of the .Net Runtime, see docs [here](https://github.com/dotnet/core/blob/master/release-notes/download-archives/2.0.2-sdk-download.md)
Do we not need to update the string to `2.1.4`, which is the SDK containing the `2.0.5` runtime, see docs [here](https://github.com/dotnet/core/blob/master/release-notes/download-archives/2.0.5-download.md)?

**Comment by iSazonov** [comment]
@bergmeister It is a trick - we use different versions of runtime and CLI. See @daxian-dbw [comment](https://github.com/PowerShell/PowerShell/pull/5295#discussion_r148621920).
`RuntimeFrameworkVersion` really set minimal patch version of .Net Core runtime.
(I found great design docs https://github.com/dotnet/designs/issues/3)

What I would really like to see is that we could start using C# 7.2 and Span/Memory (used runtime 2.0.* already contains Span support).
Also interesting when we can use .Net Core 2.1.0 - they have already done a lot to increase performance and continue to increase it.

**Pull request merged**